src/plugins/discover/public/components/discover_grid/get_render_cell_value.tsx

```diff
@@ -166,25 +180,53 @@ function renderPopoverContent({
   dataView: DataView;
   useTopLevelObjectColumns: boolean;
   fieldFormats: FieldFormatsStart;
+  closePopover: () => void;
 }) {
+  const closeButton = (
+    <EuiButtonIcon
+      aria-label={i18n.translate('discover.grid.closePopover', {
+        defaultMessage: `Close popover`,
+      })}
+      data-test-subj="docTableClosePopover"
+      iconSize="s"
+      iconType="cross"
+      size="xs"
+      onClick={closePopover}
+    />
+  );
   if (useTopLevelObjectColumns || field?.type === '_source') {
-    const json = useTopLevelObjectColumns
-      ? getInnerColumns(rowRaw.fields as Record<string, unknown[]>, columnId)
-      : rowRaw;
     return (
-      <JsonCodeEditor json={json as Record<string, unknown>} width={defaultMonacoEditorWidth} />
+      <EuiFlexGroup gutterSize="none" direction="column" justifyContent="flexEnd">
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup justifyContent="flexEnd" gutterSize="none">
+            <EuiFlexItem grow={false}>{closeButton}</EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <JsonCodeEditor
+            json={getJSON(columnId, rowRaw, useTopLevelObjectColumns)}
+            width={defaultMonacoEditorWidth}
+            height={200}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
     );
   }
 
   return (
-    <span
-      className="dscDiscoverGrid__cellPopoverValue"
-      // formatFieldValue guarantees sanitized values
-      // eslint-disable-next-line react/no-danger
-      dangerouslySetInnerHTML={{
-        __html: formatFieldValue(rowFlattened[columnId], rowRaw, fieldFormats, dataView, field),
-      }}
-    />
+    <EuiFlexGroup gutterSize="none" direction="row">
+      <EuiFlexItem>
+        <span
+          className="dscDiscoverGrid__cellPopoverValue"
```

Comment:
```suggestion
          className="dscDiscoverGrid__cellPopoverValue eui-textBreakWord"
```

Noticed that sometimes the Close button gets hidden because of a long value:
<img width="528" alt="Screenshot 2022-05-16 at 14 13 21" src="https://user-images.githubusercontent.com/1415710/168593105-3ce44a87-bcc6-4ffc-877d-bd55d4df2b64.png">